src/components/MerchantsPage/MerchantsPage.tsx

```diff
@@ -34,43 +35,52 @@ const INITIAL_CONTRIBUTIONS: ContributionsType = {
 
 ReactPixel.trackCustom('MerchantsPageView', {});
 const MerchantsPage = (props: Props) => {
+  const [loadingContrib, setLoadingContrib] = useState(false);
+  const [loadingSellers, setLoadingSellers] = useState(false);
   const websiteImages = getWebsiteImages();
   const { t, i18n } = useTranslation();
-  const [sellers, setSellers] = useState<any | null>();
-  const [filter, setFilter] = useState<any | null>();
+  const [sellers, setSellers] = useState<any[]>([]);
+  const [filter, setFilter] = useState<any[]>([]);
   const [contributions, setContributions] = useState<ContributionsType>(
     INITIAL_CONTRIBUTIONS
   );
   const [totalActiveSellers, setTotalActiveSellers] = useState<number | null>();
 
   const fetchData = async (lang?: string) => {
+    setLoadingSellers(true);
     const { data } = await getSellers(lang);
-
     setSellers(data);
     setFilter(data);
-    setContributions(
-      data.reduce(
-        (totalContributions: ContributionsType, store: BrowsePageSeller) => ({
-          donationAmount:
-            totalContributions.donationAmount + store.donation_amount,
-          giftCardAmount:
-            totalContributions.giftCardAmount + store.gift_card_amount,
-          giftAMealAmount:
-            totalContributions.giftAMealAmount + store.gift_a_meal_amount,
-        }),
-        INITIAL_CONTRIBUTIONS
-      )
-    );
     setTotalActiveSellers(
       data.filter((seller) => seller.accept_donation || seller.sell_gift_cards)
         .length
     );
+    setLoadingSellers(false);
   };
 
   useEffect(() => {
     fetchData(i18n.language);
   }, [i18n.language]);
 
+  useEffect(() => {
+    const asyncFetch = async () => {
```

Comment:
do we need another `useEffect()` hook, or can we just declare this function in the larger component and call it from the same hook? or even just place the call inside `fetchData()`? If it needs to be a separate function, let's name it something accordingly

Reply:
I prefer having separate effects mostly for clarity, one useEffect does x while the other does y rather than make one useEffect do everything. Although I'll update the function names because that's no bueno.